src/typography/Text/Text.tsx

```diff
@@ -1,27 +1,14 @@
 import React, { FC, forwardRef, Ref } from 'react';
 import classNames from 'classnames';
-import css from './Text.module.css';
-import common from '../Typography.module.css';
+import { Typography } from '../Typography';
 
-export interface TextProps extends React.HTMLAttributes<HTMLSpanElement> {
-  /**
-   * Color variation of the text.
-   */
-  color?:
-    | 'default'
-    | 'primary'
-    | 'secondary'
-    | 'error'
-    | 'success'
-    | 'warning'
-    | 'info';
+import common from '../Typography.module.css';
+import css from './Text.module.css';
 
+export interface TextProps
+  extends Typography,
+    Omit<React.HTMLAttributes<HTMLSpanElement>, 'color'> {
   fontStyle?: 'default' | 'thin' | 'bold' | 'extraBold' | 'italic';
```

Comment:
`default` should probably be `normal` to match the CSS attribute name.

Reply:
That's `fontStyle` not `fontWeight`

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-weight